### ISSUE TITLE
Parameter tuning with GridSearchCV

### ISSUE THREAD

**Issue by AlexanderGeng** [matterport/Mask_RCNN#1987]
Hello,

i use the Mask R-CNN for my project. I have to detect defects in images.
However i am not sure if my parameters (epochs, batches, optimizers, learning_rate) are suited for my problem.
I am actually doining this step by hand and try different possibilities and look what changes in accuracy.
A friend of my told me that i can try to use GridSearchCV to automaticaally select the best hyperparameters for my neural network.

I have tried to incorporate GridSearchCV but i don't know where to put exactly the code.
For an easy example like the following it works but I don't know what to do in Mask R-CNN.

[https://machinelearningmastery.com/grid-search-hyperparameters-deep-learning-models-python-keras/](url)

Can somebody help me out?

**Comment by ghost** [comment]
@Alex-259 did you manage to do this??
